testing/tests/DevExpress.ui.widgets.scheduler/currentTimeIndicator.tests.js

```diff
@@ -0,0 +1,613 @@
+            indicatorTime: new Date(2017, 8, 5, 12, 45)
+        });
+
+        var $element = this.instance.$element(),
+            $cell = $element.find(".dx-scheduler-time-panel-cell").eq(5);
+
+        assert.ok($cell.hasClass("dx-scheduler-time-panel-current-time-cell"), "Cell has specific class");
+    });
+
+    QUnit.test("TimePanel currentTime cell should have specific class, Day view with intervalCount", function(assert) {
+        this.instance.option({
+            indicatorTime: new Date(2017, 8, 6, 12, 45),
+            intervalCount: 3
+        });
+
+        var $element = this.instance.$element(),
+            $cell = $element.find(".dx-scheduler-time-panel-cell").eq(5);
+
+        assert.ok($cell.hasClass("dx-scheduler-time-panel-current-time-cell"), "Cell has specific class");
+    });
+})("DateTime indicator on Day View");
+
+(function() {
+    QUnit.module("DateTime indicator on Week View", {
+        beforeEach: function() {
+            this.instance = $("#scheduler-work-space").dxSchedulerWorkSpaceWeek({
+                showCurrentTimeIndicator: true,
+                currentDate: new Date(2017, 8, 5),
+                startDayHour: 8,
+            }).dxSchedulerWorkSpaceWeek("instance");
+        }
+    });
+
+    QUnit.test("DateTimeIndicator should be rendered if needed, Week view", function(assert) {
+        this.instance.option({
+            currentDate: new Date(),
+            startDayHour: 0
+        });
+        var $element = this.instance.$element();
+
+        assert.equal($element.find("." + SCHEDULER_DATE_TIME_INDICATOR_CLASS).length, 1, "Indicator is rendered correctly");
+
+        this.instance.option("showCurrentTimeIndicator", false);
+        $element = this.instance.$element();
+
+        assert.equal($element.find("." + SCHEDULER_DATE_TIME_INDICATOR_CLASS).length, 0, "Indicator wasn't rendered");
+    });
+
+    QUnit.test("DateTimeIndicator should not be renderd after currentDate changing, Week view", function(assert) {
+        this.instance.option({
+            indicatorTime: new Date(2017, 8, 5, 19, 45),
+            showAllDayPanel: true
+        });
+
+        var $element = this.instance.$element();
+        assert.equal($element.find("." + SCHEDULER_DATE_TIME_INDICATOR_CLASS).length, 1, "Indicator is rendered");
+
+        this.instance.option("currentDate", new Date(2017, 8, 15));
+
+        assert.equal($element.find("." + SCHEDULER_DATE_TIME_INDICATOR_CLASS).length, 0, "Indicator wasn't rendered");
+    });
+
+    QUnit.test("Shader on allDayPanel should have correct height & width, Week view", function(assert) {
+        this.instance.option({
+            indicatorTime: new Date(2017, 8, 7, 12, 45),
+            showAllDayPanel: true,
+            allDayExpanded: false
+        });
+        var $element = this.instance.$element();
+
+        assert.roughEqual($element.find("." + SCHEDULER_DATE_TIME_SHADER_ALL_DAY_CLASS).eq(0).outerHeight(), 24, 1, "Indicator has correct height");
```

Comment:
Extract `$element.find("." + SCHEDULER_DATE_TIME_SHADER_ALL_DAY_CLASS).eq(0)` to a separate variable.